pyo3-macros-backend/src/pyclass.rs

```diff
@@ -163,9 +164,17 @@ impl PyClassPyO3Options {
             };
         }
 
+        let python_version = pyo3_build_config::get().version;
+
         match option {
             PyClassPyO3Option::Crate(krate) => set_option!(krate),
-            PyClassPyO3Option::Dict(dict) => set_option!(dict),
+            PyClassPyO3Option::Dict(dict) => {
+                ensure_spanned!(
+                    python_version >= pyversions::PY_3_9 || !is_abi3(),
+                    dict.span() => "`dict` requires Python >= 3.9 with the `abi3` feature"
```

Comment:
I apologize to think of this when we've already adjusted these messages once, though I realise just now that the following might be clearer yet again. What do you think of this suggestion?

```suggestion
                    dict.span() => "`dict` requires Python >= 3.9 when using the `abi3` feature"
```

... if you like it, I think it'd be a nice final touch for the clarity of the message.

Reply:
Don't apologize. Iteration on a pr is just how things work.

Reply:
Personally, I prefer less words. Cross reference The Office meme.

But not strongly enough to object much

Reply:
Fair enough 👍  I think in which case I'd prefer to add my additional wording please given we need to touch this PR at least once more to fixup the pipeline.